sklearn/feature_extraction/dict_vectorizer.py

```diff
@@ -95,7 +96,30 @@ def __init__(self, dtype=np.float64, separator="=", sparse=True,
         self.sparse = sparse
         self.sort = sort
 
-    def fit(self, X, y=None):
+    @staticmethod
+    def _validate_types(X):
```

Comment:
You could maybe rename it to `_validate_dictvectorizer_input` and keep it outside the class?